client/src/main/java/org/eclipse/hono/client/impl/AbstractRequestResponseClient.java

```diff
@@ -1073,4 +1096,32 @@ protected final boolean isSuccessResponse(
         return StatusCodeMapper.isSuccessful(status) && payload != null
                 && MessageHelper.CONTENT_TYPE_APPLICATION_JSON.equalsIgnoreCase(contentType);
     }
+
+    /**
+     * Applies the given mapper function to the result of the given Future if it succeeded.
+     * <p>
+     * Makes sure that the given Span is finished when the given Future is completed and error information is logged
+     * to the Span if there was an error.
+     *
+     * @param resultTracker The Future supplying the <em>RequestResponseResult</em> that the mapper will be applied on.
```

Comment:
IMHO we should simply call this *result*

Reply:
I've changed it accordingly.